### PR TITLE
Changed ldd binary check for better compatibility

### PR DESCRIPTION
**Pull request by spmfox**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [ ] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**
https://github.com/pi-hole/pi-hole/issues/4697

The basic-install.sh script checks which architecture it is running on so it downloads the correct pihole-FTL binary file. It uses a simple command, `ldd /bin/ls` to determine which libraries the underlying system is using. The problem comes in when using the aarch64 container version of CentOS 8 (also RHEL 8, AlmaLinux 8, and all RHEL clones). These container images do not have 'ls' or some of the other core utilities. They use '/usr/bin/coreutils' with a argument to invoke 'ls'.

**How does this PR accomplish the above?:**
This change is simple, instead of using '/bin/ls' as the binary check - we use '/bin/bash instead'. This should not break any compatibility and after all the script is being invoked by bash.

**What documentation changes (if any) are needed to support this PR?:**
No documentation changes needed.

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

**Review by PromoFaux — DISMISSED** [review]

**Review by yubiuser — COMMENTED** [review]
Maybe use `/usr/bin/sh`? We might be more Posix compatible one day...

**Comment by spmfox** [comment]
Sure, that's what I wanted to do but I wasn't sure if /bin was preferred. I didn't want to change too much.

I'll update the pull request in a bit.

**Comment by yubiuser** [comment]
`bin` is fine

**Comment by dschaper** [comment]
Use `$(which sh)` and you don't have to worry about path.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -2216,7 +2216,7 @@ get_binary_name() {
         local rev
         rev=$(uname -m | sed "s/[^0-9]//g;")
         local lib
-        lib=$(ldd /bin/ls | grep -E '^\s*/lib' | awk '{ print $1 }')
+        lib=$(ldd /bin/bash | grep -E '^\s*/lib' | awk '{ print $1 }')
```
Comment (dschaper) [inline automated install/basic-install.sh:2219]:
```suggestion
        lib=$(ldd $(which sh) | grep -E '^\s*/lib' | awk '{ print $1 }')
```

**Review by dschaper — CHANGES_REQUESTED** [review]

**Comment by spmfox** [comment]
@dschaper I'm not sure how to update the test/test_any_automated_install.py for this change though.

**Comment by dschaper** [comment]
Okay, I think the test should be okay, but if they are not then we can mock the `ldd` call and basically echo back what ever we say the proper response should be. The test isn't about `ldd` returning a proper value but about the function knowing how to handle responses we know about and responses that we are not expecting.

So I'd test two scenarios, one that `ldd` returns the intended response (pick an architecture, doesn't matter which one really) and the other when `ldd` returns nothing or something we don't expect...

File: test/test_any_automated_install.py [hunk 2]
```diff
@@ -679,7 +679,7 @@ def test_FTL_detect_aarch64_no_errors(host):
     mock_command(
         'ldd',
         {
-            '/bin/ls': (
+            '/bin/bash': (
```
Comment (dschaper) [inline test/test_any_automated_install.py:682]:
```suggestion
            *: (
```
I don't remember offhand what the proper way to say "Return this no matter what is called"

File: test/test_any_automated_install.py [hunk 3]
```diff
@@ -807,7 +807,7 @@ def test_FTL_detect_armv8a_no_errors(host):
     # mock uname to return armv8a platform
     mock_command('uname', {'-m': ('armv8a', '0')}, host)
     # mock ldd to respond with ld-linux-armhf shared library
-    mock_command('ldd', {'/bin/ls': ('/lib/ld-linux-armhf.so.3', '0')}, host)
+    mock_command('ldd', {'/bin/bash': ('/lib/ld-linux-armhf.so.3', '0')}, host)
     detectPlatform = host.run('''
     source /opt/pihole/basic-install.sh
     create_pihole_user
```
Comment (dschaper) [inline test/test_any_automated_install.py:813]: Add a last test here that returns `No such file or directory` as the last test.

**Review by dschaper — CHANGES_REQUESTED** [review]

**Comment by spmfox** [comment]
Yeah it looks like the current test is failing, which it passed before. I don't know much python nor do I understand fully how the test even works. In my original change, I just changed 'ls' to 'bash' so it was pretty straightforward.

**Comment by dschaper** [comment]
I'm not sure if we need to mock the `which` command as well so that `which sh` returns something like `/bin/sh` and then the existing tests should behave like we expect.

File: test/test_any_automated_install.py (hunk 2 above)
Comment (dschaper) [reply test/test_any_automated_install.py:682]: Let's leave this as it is and mock `which` to return `/bin/sh` for `which sh`.

File: automated install/basic-install.sh [hunk 4]
```diff
@@ -2216,7 +2216,7 @@ get_binary_name() {
         local rev
         rev=$(uname -m | sed "s/[^0-9]//g;")
         local lib
-        lib=$(ldd /bin/ls | grep -E '^\s*/lib' | awk '{ print $1 }')
+        lib=$(ldd $(which sh) | grep -E '^\s*/lib' | awk '{ print $1 }')
```
Comment (dschaper) [inline automated install/basic-install.sh:2219]:
```suggestion
        lib=$(ldd "$(which sh)" | grep -E '^\s*/lib' | awk '{ print $1 }')
```

**Review by PromoFaux — APPROVED** [review]

**Pull request merged**